packages/firestore/src/local/shared_client_state.ts

```diff
@@ -497,26 +498,27 @@ export class WebStorageSharedClientState implements SharedClientState {
       );
 
       if (event.newValue == null) {
```

Comment:
minor, but I think this code would read better if you branch based on the key before inspecting the value.  This probably means you'll need to add a newValue != null check in your mutationBatchKeyRe branch, but I think that will actually be clearer / more explicit.